reflectable/lib/src/builder_implementation.dart

```diff
@@ -5005,7 +5005,18 @@ Future<String> _extractMetadataCode(Element element, Resolver resolver,
       if (_isPrivateName(name)) {
         await _severe('Cannot access private name $name', element);
       }
-      metadataParts.add('const $prefix$name($arguments)');
+      if (annotationNode.typeArguments != null) {
+        String typeArguments = annotationNode.typeArguments!.arguments
+            .map((TypeAnnotation typeArgument) {
+              LibraryElement library = typeArgument.type!.element!.library!;
+              String prefix = importCollector._getPrefix(library);
+              return '$prefix$typeArgument';
```

Comment:
I think this won't work in the case where the type argument is a parameterized type of the form `C<D>` where `C` is imported from `prefix1` and `D` is imported from `prefix2`: We should get `prefix1.C<prefix2.D>`, but the code above simply uses `toString()` on the given `TypeAnnotation`, and that will yield syntax which is only valid for the location where that type annotation actually occurs in the program (and at this point we are writing some code that should denote the same type, but in a different library).

Check out `_typeCodeOfTypeArgument`, you might be able to call that function in order to get the right string.

Reply:
Having some trouble calling `_typeCodeOfTypeArgument` since it is a function of `_ReflectorDomain`, which I can't seem to access from within `_extractMetadataCode`. However, it is very possible I may have missed something given that I'm not super familiar with the layout of the library.